### DISCUSSION TITLE
Get Glide URL outside of Blade

### DISCUSSION THREAD

**Discussion by MtDalPizzol** [statamic/cms#7979] [Q&A]
I'm using Inertia for my frontend which leaves me without some Statamic goodies.

One of them, is that since I'm not using Blade, I need to get the URLs for Glide images **outside of my views**, in my controllers and other points of the code before sending them to Vue to be rendered.

I couldn't find how to do this in the docs, but I can see that Statamic has an `Image` and a `Glide` `Facades`.

Can someone provide some resource that helps with that?

**Comment by duncanmcclean** [comment]
You should be able to use it in the same way you can in Blade - Blade is still PHP.

Is that not working for you?

**Reply by MtDalPizzol** [reply]
Do you mean like this?

```php
Statamic::tag('glide:generate')->src($source)->width(100)
```

**Reply by duncanmcclean** [reply]
Yes, but I think you might need `->fetch()` at the end.

**Comment by zvineyard** [comment]
I went the same route as @MtDalPizzol recommended, but you can do this the `manipulate()` method on an asset.

```php
use Statamic\Facades\Asset;

$asset = Asset::query()
	->where('container', 'assets')
	->first();

$params = [
	'w'=>'100'
];

dd($asset->manipulate($params));
```
